packages/react/src/checkbox/Checkbox.tsx

```diff
@@ -113,3 +115,8 @@ export const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
 		);
 	}
 );
+
+function useCheckboxId(idProp?: string) {
+	const autoId = useId(idProp);
+	return `checkbox-${autoId}`;
```

Comment:
Why do we need the `checkbox-` prefix?

Reply:
Just abiding by the existing pattern for all the other components. Having the name of the component in the id also helps for another user stylesheet hook as well.

Reply:
I think this may result in something unexpected if you pass in an ID prop to checkbox or radio. For example, if you pass an ID of `foo` wouldn’t the ID in the DOM now be `checkbox-foo`?

I think you want to render the ID prop if provided, otherwise fallback to `checkbox-${autoId}` , similar to these other components

https://github.com/agriculturegovau/agds-next/blob/edf33f1398b18b7ccda5ba81480e494b6d35fa47/packages/react/src/password-input/PasswordInput.tsx#L67

https://github.com/agriculturegovau/agds-next/blob/edf33f1398b18b7ccda5ba81480e494b6d35fa47/packages/react/src/combobox/utils.ts#L3

Reply:
Hmmm yeah, that would be consistent. I misread the `ControlGroup` one I stole it from. But now I'm confused. Why do we even bother passing `idProp` to `useId` if we don't use it? Now we're providing the ability to create identical ids. This code results in 2 `id="foo"` which is illegal.

```
<>
<ControlGroup
	label="Invalid example"
	message="Please choose an option"
	invalid
	block
	id="foo"
>
	<Checkbox>Phone</Checkbox>
	<Checkbox>Tablet</Checkbox>
	<Checkbox>Laptop</Checkbox>
</ControlGroup>

<ControlGroup
	label="Invalid example"
	message="Please choose an option1"
	invalid
	block
	id="foo"
>
	<Checkbox>Phone1</Checkbox>
	<Checkbox>Tablet1</Checkbox>
	<Checkbox>Laptop1</Checkbox>
</ControlGroup>
</>
```
But by not copying the existing logic, if you pass individual checkboxes and radios `id`s in a `ControlGroup`, you also get incorrect stuff:

```
<ControlGroup label="Example" block>
	<Checkbox id="foo" value="phone">Phone</Checkbox>
	<Checkbox id="foo" value="tablet">Tablet</Checkbox>
	<Checkbox id="foo" value="tablet">Laptop</Checkbox>
</ControlGroup>
```

results in `<label for="checkbox-foo"><input id="foo" />`.

So yeah, I need to do what you say, but really, why do we even care about the `id`s?

Reply:
> Why do we even bother passing idProp to useId if we don't use it?

Yeah I'm not too sure, TBH I think that is a mistake and can be updated. I _think_ you should be able to use something like the code below and have the same results.

```tsx
function usePasswordInputId(idProp?: string) {
	const autoId = useId();
	// Use ID prop if provided, otherwise fallback to auto generated ID
	return idProp || `password-input-${autoId}`;
}
```

> If you pass individual checkboxes and radios ids in a ControlGroup, you also get incorrect stuff

IMO I think that's expected behaviour. If I pass in an ID prop to a component I'd expect to see it in the DOM even though I've messed up by providing the same ID multiple times. In the code below, personally I'd expect to have the same ID in every component. You could throw a warning if multiple IDs are found on the same page, but other testing libraries should do that for the consumer.

```tsx
<Box id="foo" />
<TextInput id="foo" />
<Checkbox id="foo" />
```